server/routes/yourCompletedReports/yourCompletedReports.ts

```diff
@@ -0,0 +1,30 @@
+import { Request, Response } from 'express'
+import CompletedAdjudicationsService from '../../services/completedAdjudicationsService'
+import { pageRequestFrom, PageResponse } from '../../utils/pageResponse'
+import { CompletedAdjudicationSummary } from '../../data/CompletedAdjudicationsData'
+import mojPaginationFromPageResponse from '../../utils/pagination'
+
+export default class YourCompletedReportsRoutes {
+  constructor(private readonly completedAdjudicationsService: CompletedAdjudicationsService) {}
+
+  private renderView = async (
+    req: Request,
+    res: Response,
+    results: PageResponse<CompletedAdjudicationSummary>
+  ): Promise<void> =>
+    res.render(`pages/yourCompletedReports`, {
+      yourCompletedReports: results,
+      pagination: mojPaginationFromPageResponse(
+        results,
+        new URL(`${req.protocol}://${req.get('host')}${req.originalUrl}`)
```

Comment:
I'm probably wrong here but worth checking.. doesn't `req.originalUrl `already give you the protocol and host?

Reply:
I was hoping that was the case - but it doesn't seem to - I will investigate further though